grp/classic.gd

```diff
@@ -227,19 +236,59 @@ DeclareConstructor( "GeneralOrthogonalGroupCons",
 ##  <#/GAPDoc>
 ##
 BindGlobal( "GeneralOrthogonalGroup", function ( arg )
-  if   Length( arg ) = 2 then
+  if   Length( arg ) = 1 then
+    # form (matrix, form object, or group with stored form)
+    return GeneralOrthogonalGroupCons( IsMatrixGroup, arg[1] );
+  elif Length( arg ) = 2 and IsInt( arg[1] ) then
+    # (d, q) or (d, R)
     return GeneralOrthogonalGroupCons( IsMatrixGroup, 0, arg[1], arg[2] );
   elif Length( arg ) = 3 and IsInt(arg[1]) and IsInt(arg[2]) and
     (IsInt(arg[3]) or IsRing(arg[3])) then
+    # (e, d, q) or (e, d, R)
     return GeneralOrthogonalGroupCons( IsMatrixGroup,arg[1],arg[2],arg[3] );
+  elif Length( arg ) = 3 and IsInt(arg[1]) and
+    (IsInt(arg[2]) or IsRing(arg[2])) and
```

Comment:
Actually, it seems the possibility to pass in a ring is undocumented for GO, SO, etc. (however, it is documented for GL and SL). That seems like an oversight to me, perhaps we should also fix that (not necessarily in this PR, of course).

Reply:
I am not sure whether this was an oversight.
For `GL`, `SL`, and `Sp`, we have also methods for constructing the group in question over a residue class ring of the integers, but finite fields are the only supported rings for the orthogonal groups.

It would not be a problem to admit also `GO( 3, GF(5) )`, etc.; the documentation should then state that currently only finite fields are supported, analogous to `GL`, `SL`, and `Sp`.

Reply:
We already admit `GO(3,GF(5))`, it just is undocumented

Reply:
Perhaps the word "admit" was not a good choice.
I wrote in a commit message:
> Concerning the possibility to enter `GF(q)` instead of `q` (or more generally a ring) as an argument:
> This was already documented and supported for `Sp`, and supported but not documented for `GO` and `SO`.
> Now it will be supported but not documented also for `Omega`.
> (For `GU` and `SU`, supporting `GF(q)` (or `GF(q^2)`?) would be asking for trouble:
> The GAP library uses `q` where the Forms package expects `GF(q^2)`.)